tests/foreman/api/test_contentview.py

```diff
@@ -1167,226 +1239,262 @@ def test_positive_promote_rh_custom_spin(self):
         """Attempt to view content views
 
         @id: cdfd6e51-cd46-4afa-807c-98b2195fcf0e
 
-        @setup: create a user with the Content View read-only role
+        @setup:
 
-        @assert: User with read-only role for content view can perform all
-        Variations above
+        1. create a user with the Content View read-only role
+        2. create content view
+        3. add a custom repository to content view
 
-        @caseautomation: notautomated
+        @assert: User with read-only role for content view can view the
+        repository in the content view
 
         @CaseLevel: Integration
         """
-        # Note:
-        # Obviously all of this stuff should work with 'admin' user
-        # but these tests require creating a user with read-only permissions
-        # for Content Views
-        # THIS IS EVEN ASSUMING WE HAVE A "READ-ONLY" ROLE IN THE FIRST PLACE
-        # Dev note: none of this stuff is integrated with foreman rbac yet
-        # As such, all variations in here subject to change.
-        # Variations:
-        #  * Read, Modify,  Promote?, Publish?, Subscribe??
+        user_login = gen_string('alpha')
+        user_password = gen_string('alphanumeric')
+        # create a role with content views read only permissions
+        role = entities.Role().create()
+        entities.Filter(
+            organization=[self.org],
+            permission=entities.Permission(
+                resource_type='Katello::ContentView').search(
+                filters={'name': 'view_content_views'}),
+            role=role,
+        ).create()
+        # create read only products permissions and assign it to our role
+        entities.Filter(
+            organization=[self.org],
+            permission=entities.Permission(
+                resource_type='Katello::Product').search(
+                filters={'name': 'view_products'}),
+            role=role,
+        ).create()
+        # create a user and assign the above created role
+        entities.User(
+            organization=[self.org],
+            role=[role],
+            login=user_login,
+            password=user_password
+        ).create()
+        # Create new content view entity using admin user
+        content_view = entities.ContentView(organization=self.org).create()
+        # add repository to the created content view
+        product = entities.Product(organization=self.org).create()
+        yum_repo = entities.Repository(product=product).create()
+        yum_repo.sync()
+        content_view.repository = [yum_repo]
+        content_view = content_view.update(['repository'])
+        self.assertEqual(len(content_view.repository), 1)
+        cfg = get_nailgun_config()
+        cfg.auth = (user_login, user_password)
+        # Check that we cannot create content view due read-only permission
+        with self.assertRaises(HTTPError):
+            entities.ContentView(cfg, organization=self.org).create()
```

Comment:
It's positive test, you shouldn't check negative cases here. And this specific case is already covered in the negative test. Mind removing it?